modular_skyrat/modules/synths/code/surgery/robot_healing.dm

```diff
@@ -0,0 +1,162 @@
+//Almost copypaste of tend wounds, with some changes
+/datum/surgery/robot_healing
+	steps = list(
+		/datum/surgery_step/mechanic_open,
+		/datum/surgery_step/pry_off_plating,
+		/datum/surgery_step/cut_wires,
+		/datum/surgery_step/robot_heal,
+		/datum/surgery_step/mechanic_close,
+	)
+
+	target_mobtypes = list(/mob/living/carbon/human)
+	possible_locs = list(BODY_ZONE_CHEST)
+	replaced_by = /datum/surgery
+	requires_bodypart_type = BODYTYPE_ROBOTIC
+	surgery_flags = SURGERY_IGNORE_CLOTHES | SURGERY_REQUIRE_RESTING | SURGERY_REQUIRE_LIMB
+
+	/// The step to use in the 4th surgery step.
+	var/healing_step_type
+	/// If true, doesn't send the surgery preop message. Set during surgery.
+	var/surgery_preop_message_sent = FALSE
+
+/datum/surgery/robot_healing/New(surgery_target, surgery_location, surgery_bodypart)
+	..()
+	if(healing_step_type)
+		steps = list(
+			/datum/surgery_step/mechanic_open,
+			/datum/surgery_step/pry_off_plating,
+			/datum/surgery_step/cut_wires,
+			healing_step_type,
+			/datum/surgery_step/mechanic_close,
+		)
+
+/datum/surgery_step/robot_heal
+	name = "repair body (welder/cable)"
+	implements = list(TOOL_WELDER = 100, /obj/item/stack/cable_coil = 100)
+	repeatable = TRUE
+	time = 25
+
+	/// If this surgery is healing brute damage. Set during operation steps.
+	var/heals_brute = FALSE
+	/// If this surgery is healing burn damage. Set during operation steps.
+	var/heals_burn = FALSE
+	/// How much healing the sugery gives.
+	var/brute_heal_amount = 0
+	/// How much healing the sugery gives.
+	var/burn_heal_amount = 0
+	/// Heals an extra point of damage per X missing damage of type (burn damage for burn healing, brute for brute). Smaller number = more healing!
+	var/missing_health_bonus = 0
+
+/datum/surgery_step/robot_heal/tool_check(mob/user, obj/item/tool)
+	if(implement_type == TOOL_WELDER && !tool.tool_use_check(user, 1))
+		return FALSE
+	return TRUE
```

Comment:
This feels wrong, no? Like, this means it's always going to return TRUE if you're using anything but a welder, no? I'm not super knowledgeable in surgery code, mind you.

Reply:
Oh my fucking god you're correct

Reply:
Wait no. Implement type is set when player clicks on patient, so that's completely fine.

Reply:
It assumes the supercheck did it's job for wires... which is super lazy, but understandable in it's own dumb way.